src/user.py

```diff
@@ -443,23 +443,23 @@ def user_update(
     # Get modifications from arguments
     new_attr_dict = {}
     if firstname:
-        new_attr_dict["givenName"] = [firstname]  # TODO: Validate
-        new_attr_dict["cn"] = new_attr_dict["displayName"] = [
-            (firstname + " " + user["sn"][0]).strip()
-        ]
+        new_attr_dict["givenName"] = firstname  # TODO: Validate
+        new_attr_dict["cn"] = new_attr_dict["displayName"] = (
+            firstname + " " + user["sn"][0]
+        ).strip()
```

Comment:
There are similar stuff in `portal.py`'s `portal_update()` that need to be adapted too